crates/sui-graphql-client/src/lib.rs

```diff
@@ -426,6 +428,38 @@ impl Client {
         }
     }
 
+    /// The total number of transaction blocks in the network by the end of the provided checkpoint.
+    /// Only one of `checkpoint_digest` or `checkpoint_seq_num` should be provided. If none is
+    /// provided, it will return the total number of transactions by the end of the last known
+    /// checkpoint.
+    pub async fn total_transaction_blocks(
+        &self,
+        digest: Option<String>,
+        seq_num: Option<u64>,
+    ) -> Result<Option<u64>, Error> {
+        ensure!(
+            !(digest.is_some() && seq_num.is_some()),
+            "Either digest or seq_num must be provided"
```

Comment:
This message conflicts with your doc comment. Its either a digest or a sequence number or neither.

Also this type of api isn't really that great as it forces you to always provide "None" to at least one of the params. Also the digest should likely be strongly typed vs stringly typed.